rs/tests/execution/general_execution_tests/nns_shielding.rs

```diff
@@ -141,6 +142,103 @@ pub fn mint_cycles_not_supported_on_application_subnet(env: TestEnv) {
     });
 }
 
+pub fn mint_cycles128_supported_only_on_cycles_minting_canister(env: TestEnv) {
+    let nns_node = env.get_first_healthy_nns_node_snapshot();
+    let specified_id = nns_node.get_last_canister_id_in_allocation_ranges();
+    // Check that 'specified_id' is not 'CYCLES_MINTING_CANISTER_ID'.
+    assert_ne!(specified_id, CYCLES_MINTING_CANISTER_ID.into());
+    let nns_agent = nns_node.build_default_agent();
+    block_on(async move {
+        let not_cmc = UniversalCanister::new_with_cycles(&nns_agent, specified_id, *INITIAL_CYCLES)
+            .await
+            .unwrap()
+            .canister_id();
+
+        let before_balance = get_balance(&not_cmc, &nns_agent).await;
+        assert_balance_equals(
+            *INITIAL_CYCLES,
+            Cycles::from(before_balance),
+            BALANCE_EPSILON,
+        );
+
+        let res = nns_agent
+            .update(&not_cmc, "update")
+            .with_arg(
+                wasm()
+                    .mint_cycles128(Cycles::from((1u128 << 64) + 2u128))
```

Comment:
we could try to mint a small amount here to ensure that this fails because of the canister id (rather than due to too many cycles)